### PR TITLE
replace deps encoding with build in textdecoder

### PR DESCRIPTION
**Pull request by Uzlopak**
We should discuss about this first. This replaces the jsencoding code with the built in textdecoder. 

We have unfortunately no busboy benchmarks. Maybe we should add first some benchmarks before checking the validity of this PR.

I cache the instantiated TextDecoder in a Map because I expect that instantiating a TextDecoder is expensive and reusing it is cheap and it has a low memory footprint. Maybe my expectation is wrong?

Nice sideeffect, the test coverage increases massively. LOL
```
File               | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s                                             
-------------------|---------|----------|---------|---------|---------------------------------------------------------------
All files          |   93.15 |    85.52 |   95.38 |   93.03 |                                                               
```

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by kibertoad** [comment]
I second the idea of having busboy benchmarks, we could do an informed decision here then.

File: lib/utils.js [hunk 1]
```diff
@@ -94,13 +92,34 @@ function parseParams(str) {
...
-  if (text && jsencoding.encodingExists(destEncoding)) {
-    try {
-      ret = jsencoding.TextDecoder(destEncoding)
-                      .decode(Buffer.from(text, textEncoding));
-    } catch(e) {}
+  let ret;
+  /**
+   * https://datatracker.ietf.org/doc/html/rfc7578#section-4.7
+   */
+  switch (destEncoding) {
```
Comment (kibertoad) [inline lib/utils.js:102]: no default? what will hapoen if we don't resolve anything there?

File: lib/utils.js [hunk 2]
```diff
@@ -94,13 +92,34 @@ function parseParams(str) {
...
+    case '7bit':
+      destEncoding = 'ascii';
+      break;
+    case '8bit':
+      destEncoding = 'binary';
+      break;
+  }
+  if (text) {
```
Comment (kibertoad) [inline lib/utils.js:110]: do we need destEncoding ourmtside of this block?

File: lib/utils.js (hunk 1 above)
Comment (kibertoad) [inline lib/utils.js:102]: if we only support two encodings, do we even need map? can have two consts and switch between them.

File: lib/utils.js [hunk 3]
```diff
@@ -94,13 +92,34 @@ function parseParams(str) {
...
-      ret = jsencoding.TextDecoder(destEncoding)
-                      .decode(Buffer.from(text, textEncoding));
-    } catch(e) {}
+  let ret;
+  /**
+   * https://datatracker.ietf.org/doc/html/rfc7578#section-4.7
+   */
+  switch (destEncoding) {
+    case '7bit':
+      destEncoding = 'ascii';
```
Comment (kibertoad) [inline lib/utils.js:104]: do we mutate arg instead of introducing new const for perf reasons?

**Comment by kibertoad** [comment]
I am very scared of regressing edge cases here, considering both the size of original implementation and low code coverage of it. Wonder if we could transitively increase code coverage of encoder via writing additional tests for busboy itself first.

File: lib/utils.js (hunk 2 above)
Comment (Uzlopak) [reply lib/utils.js:111]: no, we get from outside if it is 7bit or 8bit. But I think I made a mistake because this is actually the textEncoding and not the destEncoding

Comment (kibertoad) [reply lib/utils.js:111]: considering that all tests still passed, it's a very alarming sign. we do need to improve our code coverage for this part of functionality ahead of this change.

**Comment by Uzlopak** [comment]
I think that jsencoding dependency is actually an archaic TextDecoder Implementation. 

But what i realized is, that if I output the error messages of the created try catch blocks, it actually throws errors in node 10 and 12 reporting that iso-8891-1, despite the unit test passes. Maybe because the files are utf-8 encoded?

```
    ✔ Multiple extended parameters (RFC 5987) with mixed charsets
RangeError [ERR_ENCODING_NOT_SUPPORTED]: The "iso-8859-1" encoding is not supported
```

So I think, the issue is, that the built in TextDecoder is depending on the built-in ICU. 
https://nodejs.org/docs/latest-v14.x/api/util.html#util_encodings_supported_when_icu_is_disabled

So what we can do is, to try to get a built-in TextDecoder and if we throw we use the deps/encoding like before as fallback. Maybe if we split the encoders, and require them if necessary reduces the start up time of busboy. But only if it means, that we have some performance gains. 

So we should postpone this PR till we have some busboy benchmarks.

File: lib/utils.js (hunk 3 above)
Comment (Uzlopak) [reply lib/utils.js]: removed that

File: lib/utils.js (hunk 1 above)
Comment (Uzlopak) [reply lib/utils.js]: not necessary :)

File: lib/utils.js (hunk 2 above)
Comment (Uzlopak) [reply lib/utils.js]: I think it would be a huge task to do that, with little to no benefits. Or are there any nodejs unit tests for the builtin textdecoder, which we could reuse?

**Comment by Uzlopak** [comment]
@kibertoad We could remove actually all encoders, as we dont need the encoders but only the decoders.

**Comment by Uzlopak** [comment]
I removed the encoders and extracted the big json array into encodings.json.

**Comment by Uzlopak** [comment]
I tested it with the benchmarks from #40 and it is surprisingly slower. So I would say, We should not use the built-in TextDecoder...
